Fix JS error after add to cart

iqit-commerce opened the pull request:
After add to cart there was JS error in console which was blocking refresh of product page

<!-- Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows with the necessary information: -->

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.4.x
| Description?  | Fix js issue after add to cart https://pasteboard.co/HvLuX0p.png
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | go to product page, and add to cart, click contiune and check chrome debug javascript console

<!-- Click the form's "Preview button" to make sure the table is functional in GitHub. Thank you! -->

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/9364)
<!-- Reviewable:end -->

iqit-commerce committed:
Fix JS error after add to cart

After add to cart there was JS error in console which was blocking refresh of product page

PierreRambaud commented on themes/classic/_dev/js/components/block-cart.js:
```diff
@@ -42,7 +42,7 @@ prestashop.blockcart.showModal = (html) => {
   $blockCartModal = getBlockCartModal();
   $blockCartModal.modal('show').on('hidden.bs.modal', (event) => {
     prestashop.emit('updateProduct', {
-      reason: event.currentTarget.dataset
```
Not sure the reason should be deleted. But nice catch for event.

Quetzacoalt91 replied:
That's right, reason is kept for backward compatibility with < v1.7.4.0 themes

iqit-commerce replied:
Ok, I made another commit to bring reason back

iqit-commerce committed:
Fix JS error after add to cart

After add to cart there was JS error in console which was blocking refresh of product page

PierreRambaud commented:
@Quetzacoalt91 Not sure QA can reproduce due to the fact there is already another fix on updateProduct function :) We can merge without QA, wdyt?

Quetzacoalt91 commented:
Even if there is another fix in progress, do you really think we cannot check the error reported by @iqit-commerce?

PierreRambaud commented:
We can't reproduce because of another fix. We will not see any errors in the console even if his fix is legit :/

Quetzacoalt91 commented:
You can assign it to the QA team anyway, and let them decide if they can check it or try to find a regression somewhere. They will see your comment.

Quetzacoalt91 commented:
Thank you @iqit-commerce
